src/main/scala/com/gu/configraun/models/Stage.scala

```diff
@@ -1,7 +1,18 @@
 package com.gu.configraun.models
 
-sealed trait Stage { def name: String }
+object Stage {
+  def fromString(s: String) = Set(DEV, CODE, PROD).find(a => a.name.equals(s))
+}
+
+sealed trait Stage {
+  def name: String
+}
+
 case object DEV extends Stage { def name: String = "DEV" }
 case object CODE extends Stage { def name: String = "CODE" }
 case object PROD extends Stage { def name: String = "PROD" }
 
+class Stack(val value: String) extends AnyVal
```

Comment:
This can be a `case class`. Case classes can still be used for value classes and upon instantiation then you won't be required to use the `new` keyword. This is more idiomatic Scala.